modules/loans/src/lib.rs

```diff
@@ -194,8 +196,18 @@ impl<T: Config> Pallet<T> {
 		let module_account = Self::account_id();
 
 		if collateral_adjustment.is_positive() {
+			ensure!(
+				T::Currency::total_balance(currency_id, who).saturating_sub(collateral_balance_adjustment)
+					>= T::Currency::minimum_balance(currency_id),
+				Error::<T>::BalanceTooLow
+			);
 			T::Currency::transfer(currency_id, who, &module_account, collateral_balance_adjustment)?;
 		} else if collateral_adjustment.is_negative() {
+			ensure!(
```

Comment:
why do we need those checks? we can just let `transfer` to fail?
if user already have other assets that provides provider, we should allow reduce this asset to zero